### PR TITLE
General Updates 12

### LINKED ISSUE #182: Contribution graph not displaying correctly after GU 11
### Before you start...

- [X] Have you updated FluentHub? You might be using an old version.
- [X] Have you checked if a similar issue has already been reported?

### What version of Windows is this issue present?

Windows 11 Dev

### What version/architecture of FluentHub are you on?

0.3.0.0 | X64 | DEBUG

### Description

Contribution graph under "Your Profile" is not displaying properly, or at least how it used to before GU 11.  Get three days instead of entire calendar.

### Steps To Reproduce

1. Log in/open
2. Select profile picture in top right corner
3. Select "Your Profile"

### Expected behavior

image

### Relevant Assets

image

image

### PR DESCRIPTION
**Pull request by onein528**
### Before you start
- [x] Have you checked if a similar PR has already been requested? 
- [x] Have you built and ran the app?

### Related/Fixed issues

- Closes #182

### Description

Official General Update round 12

#### Todo

- [ ] Add click event to activities button in activities section in home page.
- [ ] Add click event to feed items in activities page.
- [ ] Handle OAuth Restriction exception properly.
- [ ] Completely update and improve indivisual issue/PR page.
- [ ] Add repository Insights page
- [ ] Add organization Peoples page
- [ ] Add organization Projects page
- [ ] Add filtering feature (reposityries, users, notifications, ...)
- [ ] Update indivisual Discussion page
- [ ] Update indivisual Project page
- [ ] Add repository Actions page
- [ ] Add search page

#### Bugs known

_None_

### Assets (if appropriate):

Redesigned sign in page

image

### CODE REVIEW

**Comment by onein528** [comment]
image

We can like this? [MyFTP](https://github.com/luandersonn/MyFTP) using muxc:TwoPaneView
Code https://github.com/luandersonn/MyFTP/blob/main/MyFTP/Views/LoginViewPage.xaml#L52-L259

**Comment by DeveloperWOW64** [comment]
> We can like this? [MyFTP](https://github.com/luandersonn/MyFTP) using muxc:TwoPaneView Code https://github.com/luandersonn/MyFTP/blob/main/MyFTP/Views/LoginViewPage.xaml#L52-L259

@onein528 for login? if so, yes! I love the look

**Comment by onein528** [comment]
@DeveloperWOW64 Yes. problem is we have no image fitting this pane mode.

**Comment by DeveloperWOW64** [comment]
close accidental?

**Comment by onein528** [comment]
yep.

**Comment by DeveloperWOW64** [comment]
oops

**Comment by DeveloperWOW64** [comment]
> @DeveloperWOW64 Yes. problem is we have no image fitting this pane mode.

I can generate one later

**Comment by DeveloperWOW64** [comment]
@onein528 Done! I can't code now but heres the image:
FluentHub Hero v2
It's exactly the same size as the one in MyFTP.

**Comment by onein528** [comment]
Thanks!!

**Comment by DeveloperWOW64** [comment]
ill dev now

**Comment by DeveloperWOW64** [comment]
> Shall we use StyledOction for primer/octions. octions does not expect to use as font(abandoned long time ago). And so we should have as svg and handle those svg in StyledOction user control.
> 
> StuledOction is one of the github's design.

It will take time, but sure. @onein528, you do realise it's called _Octicons_ and not _Octions_

**Comment by onein528** [comment]
wow 'c' missed. haha thank you ;)

**Comment by onein528** [comment]
@DeveloperWOW64 @BobbyESP @luandersonn  

Do you guys have any ideas other than to merge this?

**Comment by DeveloperWOW64** [comment]
> Do you guys have any ideas other than to merge this?

@onein528 what's changed?

and what about these:
image

**Comment by onein528** [comment]
@DeveloperWOW64 Yes. I updated No.4 to some extent.

Other that those goals. ideas, or something.

**Comment by DeveloperWOW64** [comment]
> @DeveloperWOW64 Yes. I updated No.4 to some extent.
> 
> Other that those goals. ideas, or something.

okay. sure then, you can merge. ill review for you

**Review by DeveloperWOW64 — DISMISSED** [review]

**Comment by onein528** [comment]
No. I dont want to merge. if you have any ideas of something, you or I can add that feature.

**Comment by DeveloperWOW64** [comment]
> No. I dont want to merge. if you have any ideas of something, you or I can add that feature.

hmm.

**Comment by DeveloperWOW64** [comment]
Add click event to activities button in activities section in home page.

**Comment by DeveloperWOW64** [comment]
> Add click event to activities button in activities section in home page.

that should be easy, should it not?

**Comment by DeveloperWOW64** [comment]
Also i wish we could have this:
image
but it relies on certain WPF controls and FH is UWP

**Comment by onein528** [comment]
oh, right. do you have an thought about other UIs?

**Comment by DeveloperWOW64** [comment]
> oh, right. do you have an thought about other UIs?

MS Get started

**Comment by onein528** [comment]
Im sorry. not meant to. I want.. ah

Do you have any new UI concepts or ideas other than the intro page like repo page, issue page?

**Comment by DeveloperWOW64** [comment]
> Im sorry. not meant to. I want.. ah
> 
> Do you have any new UI concepts or ideas other than the intro page like repo page, issue page?

no actually

**Comment by DeveloperWOW64** [comment]
I saw this concept tho - it could be used in Insights
image

**Comment by onein528** [comment]
wow it's very cool!! thanks!

**Comment by DeveloperWOW64** [comment]
I'm terrible at backend so I could do some UI if you want

**Comment by DeveloperWOW64** [comment]
@onein528 I'll dev in a bit - sry i've got other things to do

**Comment by DeveloperWOW64** [comment]
@onein528 what's the difference between using Octokit.GraphQL.NET and Octokit.Net?

**Comment by onein528** [comment]
Response speed. The second one allows me to get all data despite it is large amount. But first one allows me to get data specifically. For example, when I want to get specific repository owner only, it allows me.

**Comment by DeveloperWOW64** [comment]
> Response speed. The second one allows me to get all data despite it is large amount. But first one allows me to get data specifically. For example, when I want to get specific repository owner only, it allows me.

ah okay

**Comment by DeveloperWOW64** [comment]
Also, I'm not so sure Octokit.NET and Octokit.GraphQL.NET are abandonded - they both have recent commits

**Comment by DeveloperWOW64** [comment]
I feel i should dev as I have time but I do not know what to work on.

**Comment by DeveloperWOW64** [comment]
@onein528 Why don't we use strings?

**Comment by onein528** [comment]
due to massive UI updates.

**Comment by DeveloperWOW64** [comment]
> due to massive UI updates.

how are they related?

**Comment by onein528** [comment]
We had localization resources until GU2 was committed, but since GU3, the UI has noticeably changed a lot and the Uid has become more difficult to manage. We removed the resource because we often had to delete the resource and create a new one soon after we had it translated to Japanese, Spanish, or Chinese.

**Comment by DeveloperWOW64** [comment]
I think I may have found a way to code the MyFTP login screen.

**Comment by DeveloperWOW64** [comment]
> I think I may have found a way to code the MyFTP login screen.

nope. It works but it is very ugly and not like what I want it to be

**Comment by DeveloperWOW64** [comment]
@onein528 Are you deving?

**Comment by onein528** [comment]
Finally

image

**Comment by onein528** [comment]
> @onein528 Done! I can't code now but heres the image: FluentHub Hero v2 It's exactly the same size as the one in MyFTP.

Actually, I dont think this hero image is good. kind of "image" is good like MyFTP using.

**Comment by DeveloperWOW64** [comment]
@onein528 why?

**Comment by onein528** [comment]
ah, window in window image is not good I think.

**Comment by DeveloperWOW64** [comment]
> ah, window in window image is not good I think.

why?

**Comment by onein528** [comment]
Nice hero. but little fluent

image

I dont know what is better. but MyFTP like image is good I think:

vector1

**Comment by DeveloperWOW64** [comment]
Love the second one

**Comment by onein528** [comment]
Really. But i dont know it is legal to use.

**Comment by DeveloperWOW64** [comment]
> Really. But i dont know it is legal to use.

Where'd you get it?

**Comment by DeveloperWOW64** [comment]
If it's from shutterstock, 100% attribute. I don't want an app with an attribute.
My wallpaper library has over 75 wallpapers though.. Any wallpaper related to Windows would not need attribution as the user would be running the OS. Your profile pic, e.g.
image

**Comment by DeveloperWOW64** [comment]
@onein528 What do you think?

**Comment by onein528** [comment]
Pick one up please.

(What's the "attribute"?)

**Comment by DeveloperWOW64** [comment]
> Pick one up please.
> 
> (What's the "attribute"?)

Attribution is when you need to say that you have obtained an asset (this case: image) from a place, rather than just use it.

Here is one of them:
Surface_Laptop_3_07

**Comment by onein528** [comment]
you like this?

>

**Comment by DeveloperWOW64** [comment]
Yes. This is one of the Microsoft wallpapers, designed by the same team as WinUI i think, so it should fit well.

**Comment by onein528** [comment]
Do you think?

image
image

**Comment by DeveloperWOW64** [comment]
WOAH That looks fantastic!!

**Comment by DeveloperWOW64** [comment]
@onein528 Can you push?

**Comment by onein528** [comment]
Yep. a sec.

**Comment by DeveloperWOW64** [comment]
@onein528 hmm..
image

**Comment by DeveloperWOW64** [comment]
ありがとう！

**Comment by DeveloperWOW64** [comment]
@onein528 What do you want to complete before the merge of this PR? These?
image

**Comment by onein528** [comment]
first 3.

**Comment by onein528** [comment]
@DeveloperWOW64 Do you or others know something good app or web to create todo, milestone, or loadmap?

**Comment by DeveloperWOW64** [comment]
> @DeveloperWOW64 Do you or others know something good app or web to create todo, milestone, or loadmap?

@onein528 GitHub Projects is actually best because it is integrated so well. Projects also allows for 'iterations' - lots of PRs and issues linked to one feature/bug.

**Comment by onein528** [comment]
Ok. I will take a look at docs written how to use that.

**Comment by DeveloperWOW64** [comment]
> Ok. I will take a look at docs written how to use that.

you want to use it for FH? I can set it up.

**Comment by onein528** [comment]
yes. I have lots of To-do and its milestone, loadmap

**Comment by onein528** [comment]
v0.0 - v0.5 Done querying feature implementing(read)
v0.5 - v0.9 Done mutation feature implementing(edit)
v0.9 - v1.0 Prepare to publish stable version

**Comment by DeveloperWOW64** [comment]
Ok. I'll create these milestones - I might have to delete the current project tho

**Comment by DeveloperWOW64** [comment]
@onein528 This any good?
https://github.com/orgs/FluentHub/projects/4

**Comment by DeveloperWOW64** [comment]
@onein528 I tried to revert these because of the way GH Projects works. Once we're done with Query the Current channel will become mutation, even though it's called query.
image

**Comment by onein528** [comment]
Oh sorry. please revert it. I've changed.

**Comment by DeveloperWOW64** [comment]
done

**Comment by DeveloperWOW64** [comment]
@onein528 What do you mean by mutation?

**Comment by onein528** [comment]
refer to https://docs.github.com/en/graphql/reference/mutations

**Comment by DeveloperWOW64** [comment]
> refer to https://docs.github.com/en/graphql/reference/mutations

So the user can actually do stuff other than browse GH?

**Comment by onein528** [comment]
yep. edit user info, submit issue, add comment...

**Comment by DeveloperWOW64** [comment]
Cool! if you have any more features for planning, tell me.

**Comment by onein528** [comment]
WDYM??

**Comment by DeveloperWOW64** [comment]
> WDYM??

e.g. you want to add new feature that isn't one of the big 3.

**Comment by onein528** [comment]
hahah hhaaa.....

image

**Comment by DeveloperWOW64** [comment]
> hahah hhaaa.....

bruh

**Comment by onein528** [comment]
No, I will pick up from those.

**Comment by onein528** [comment]
1,2,3,4,5,6 Actually.

From those list you can also open and merge a PR.

**Comment by DeveloperWOW64** [comment]
> From those list you can also open and merge a PR.
> 
> 1,2,3,5,6,7 Actually.

WDYM?

**Comment by onein528** [comment]
You can commit more than one from those.

image

I want to commit No 1,2,3,5,6,7 into this.

**Comment by DeveloperWOW64** [comment]
> I want to commit No 1,2,3,5,6,7 into this.

You deving now?

**Comment by onein528** [comment]
yep. adding insights page.

**Comment by DeveloperWOW64** [comment]
> yep. adding insights page.

With the design I said or a different one?

**Comment by onein528** [comment]
but will get slow down another two weeks. please keep in mind.

**Comment by DeveloperWOW64** [comment]
also, just to be clear, are you starting on mutation/query in this PR?

**Comment by onein528** [comment]
> > yep. adding insights page.
> 
> With the design I said or a different one?

I dont know how it goes, but almost follow the one what you said. I really hate current GitHub Insights UI

**Comment by onein528** [comment]
> also, just to be clear, are you starting on mutation/query in this PR?

No. delete that.

**Comment by DeveloperWOW64** [comment]
> > also, just to be clear, are you starting on mutation/query in this PR?
> 
> No. delete that.

delete what?

**Comment by onein528** [comment]
from todo list

**Comment by DeveloperWOW64** [comment]
I'm working on the GH project - adding stuff, etc.

**Comment by onein528** [comment]
UI doesn't work properly despite I added UIs

**Comment by onein528** [comment]
Do you know why?

**Comment by DeveloperWOW64** [comment]
> Do you know why?

sry just building..

**Comment by DeveloperWOW64** [comment]
@onein528 hmmm (again)
image

**Comment by onein528** [comment]
No way. can you clean solution and rebuild?

**Comment by DeveloperWOW64** [comment]
> No way. can you clean solution and rebuild?

nope. still doesn't work.

**Comment by onein528** [comment]
Can you debug?

**Comment by DeveloperWOW64** [comment]
> Can you debug?

ofc thats what Im doing. debugging is just reporting where the error is when you get it, instead of just crashing

**Comment by onein528** [comment]
so can you track why token is empty? I cannot track because this exception does not occur on my end.

**Comment by DeveloperWOW64** [comment]
> so can you track why token is empty? I cannot track because this exception does not occur on my end.

I know some, but I'm very bad at C#. Maybe it's because my XML document has wrong format.
Is this the same format as yours?
image

**Comment by DeveloperWOW64** [comment]
@onein528 Fixed it. Just some weird error. I reinstalled FH, which did the trick.

**Comment by onein528** [comment]
FluentHub is sometimes strange.

**Comment by DeveloperWOW64** [comment]
May have fixed the UI broken issue with [`d13045a`](https://github.com/FluentHub/FluentHub/pull/180/commits/d13045a0cf6150e2533a565d1f9a9f0f7a85f50a)

**Comment by DeveloperWOW64** [comment]
@onein528 now we need UI. Maybe we could use GridView?

**Comment by onein528** [comment]
> now we need UI. Maybe we could use GridView?

Talk about where?

And did you fixed insights page?

**Comment by DeveloperWOW64** [comment]
> > now we need UI. Maybe we could use GridView?
> 
> Talk about where?
> 
> And did you fixed insights page?

I think I fixed insights page - wdym by 'talk about where?'

**Comment by onein528** [comment]
> now we need UI.

What do you mean? talking about insighs page UI?

> I think I fixed insights page

Can you give me a screenshot of that?

**Comment by DeveloperWOW64** [comment]
> Can you give me a screenshot of that page?

@onein528 
image

**Comment by onein528** [comment]
On mine, literally broken.

image

**Comment by DeveloperWOW64** [comment]
> On mine, literally broken.

Pull from origin @onein528

**Comment by onein528** [comment]
Wow nice.

**Comment by onein528** [comment]
image

**Comment by DeveloperWOW64** [comment]
I'll be able to dev in a bit

**Comment by onein528** [comment]
Going to sleep. bye.

**Comment by DeveloperWOW64** [comment]
> Going to sleep. bye.

さよなら ともゆき。

**Comment by DeveloperWOW64** [comment]
Definitely broke fluenthub..
image

**Comment by onein528** [comment]
I will create sign out methods. Please revert “reset” arg implemented commit. Cannot do like that.

**Comment by onein528** [comment]
```cs
Frame rootFrame = (Frame)Window.Current.Content;
rootFrame.Navigate(typeof(SignIn.IntroPage));
```

This code does not work?

**Comment by DeveloperWOW64** [comment]
> ```cs
> Frame rootFrame = (Frame)Window.Current.Content;
> rootFrame.Navigate(typeof(SignIn.IntroPage));
> ```
> 
> This code does not work?

no.

**Comment by onein528** [comment]
@luandersonn Do you know what kinda sliding animation is? In the situation where you want to change the content from “1” to “2”. Slide top to bottom.

**Comment by onein528** [comment]
2 comes from above and 1 go down

**Comment by DeveloperWOW64** [comment]
> @luandersonn Do you know what kinda sliding animation is? In the situation where you want to change the content from “1” to “2”. Slide top to bottom.

@onein528 winui3gallery://item/PageTransition or if you don't have winui 3 gallery, https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.Media.Animation.NavigationThemeTransition

**Comment by onein528** [comment]
Not page transition animation.

**Comment by DeveloperWOW64** [comment]
> Not page transition animation.

oh. as in content? It must be one of these:
image

**Comment by onein528** [comment]
@DeveloperWOW64 Are you really not good at C#?

**Review by ioapiset — APPROVED** [review]
LGTM.

**Comment by onein528** [comment]
Actions at work.

**Comment by DeveloperWOW64** [comment]
> @DeveloperWOW64 Are you really not good at C#?

Not amazing, not terrible..

**Comment by DeveloperWOW64** [comment]
@onein528 Why the review?

**Comment by DeveloperWOW64** [comment]
o no maybe i broke it with [`c5b7fc5`](https://github.com/FluentHub/FluentHub/pull/180/commits/c5b7fc53cc05150496dee8def1d604fca9212785)

File: src/FluentHub/FluentHub.csproj [hunk 1]
```diff
@@ -17,9 +17,18 @@
...
+    <GenerateAppInstallerFile>True</GenerateAppInstallerFile>
+    <PackageCertificateThumbprint>64B7A4274C47A261D02A8672BA06EEBD7926E47E</PackageCertificateThumbprint>
+    <AppxPackageSigningTimestampDigestAlgorithm>SHA256</AppxPackageSigningTimestampDigestAlgorithm>
+    <AppxAutoIncrementPackageRevision>True</AppxAutoIncrementPackageRevision>
+    <GenerateTestArtifacts>True</GenerateTestArtifacts>
+    <AppxBundle>Always</AppxBundle>
+    <AppxBundlePlatforms>x86|x64|arm</AppxBundlePlatforms>
+    <AppInstallerUri>https://api.codrex.dev/_get/_file/%24context=fluenthub/</AppInstallerUri>
+    <HoursBetweenUpdateChecks>0</HoursBetweenUpdateChecks>
   </PropertyGroup>
```
[the remark this answers is not in the document]
Comment (DeveloperWOW64) [reply src/FluentHub/FluentHub.csproj:32]: why? @onein528  they are required to generate the debug apppackages

[the remark this answers is not in the document]
Comment (onein528) [reply src/FluentHub/FluentHub.csproj:32]: then why did you allowed including pfx files?

**Comment by DeveloperWOW64** [comment]
> Changes summary:
> 
> ```
> + Added FUNDING.yml
> + Updated README.md
> + Added some schemas of Octokit.GraphQL
> + Added some url handling
> + Updated intro page
> + Added hero image
> + Improved some pages' code behind and those viewmodels
> + Added url handing to each page
> + Added some repo's insights page
> + Improved indivisual issue/PR page
> + Updated issue/PR's timeline
> ```
> 
> @DeveloperWOW64 Why did you add .pfx file?

certificate for apppackage debug. it will fail without it.

**Comment by onein528** [comment]
I see.

File: src/FluentHub/FluentHub.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (DeveloperWOW64) [reply src/FluentHub/FluentHub.csproj:32]: > then why did you allowed including pfx files?

so it would build. it fails without.

**Comment by onein528** [comment]
`No certificate found with the supplied thumbprint: 64B7A4274C47A261D02A8672BA06EEBD7926E47E`

Can you fix build errors?

**Comment by DeveloperWOW64** [comment]
> `No certificate found with the supplied thumbprint: 64B7A4274C47A261D02A8672BA06EEBD7926E47E`
> 
> Can you fix build errors?

should work with pfx file.

**Comment by onein528** [comment]
See latest pipeline result https://github.com/FluentHub/FluentHub/pull/180/checks?check_run_id=7200590807

**Comment by onein528** [comment]
on your end, it will success?

**Comment by DeveloperWOW64** [comment]
> on your end, it will success?

yes

**Comment by onein528** [comment]
failed with same error.

**Comment by DeveloperWOW64** [comment]
@onein528 I will try on my machine.

**Comment by onein528** [comment]
And can you fix my grammer:?

```
+ Added FUNDING.yml
+ Updated README.md
+ Added some schemas of Octokit.GraphQL
+ Added some url handling
+ Updated intro page
+ Added hero image
+ Improved some pages' code behind and those viewmodels
+ Added url handing to each page
+ Added some repo's insights page
+ Improved indivisual issue/PR page
+ Updated issue/PR's timeline
```

**Comment by DeveloperWOW64** [comment]
Fixed:

```
+ Added FUNDING.yml
+ Updated README.md
+ Added schemas for Octokit.GraphQL
+ Added better support for URL handling
+ Updated Out of Box Experience
+ Added hero image (OOBE)
+ Improved some pages' code-behind and their ViewModels
+ Added Insights page
+ Improved individual issue/PR pages
+ Updated timeline of events in issues/PRs
```
@onein528

**Comment by DeveloperWOW64** [comment]
> failed with same error.

Weird. @onein528 My one works.
image

**Comment by onein528** [comment]
Can you revert c5b7fc53cc05150496dee8def1d604fca9212785 and 1360e59b08146e6804907e57a3f8b80e3d279f87 ?

**Comment by DeveloperWOW64** [comment]
> Can you revert [c5b7fc5](https://github.com/FluentHub/FluentHub/commit/c5b7fc53cc05150496dee8def1d604fca9212785) and [1360e59](https://github.com/FluentHub/FluentHub/commit/1360e59b08146e6804907e57a3f8b80e3d279f87) ?

sure.

**Comment by onein528** [comment]
When generate appx packages, always packag manager create those capabilities and pfx so i think dont need.

**Comment by DeveloperWOW64** [comment]
@onein528 Done with https://github.com/FluentHub/FluentHub/pull/180/commits/9b7b7adc813b92395f2165a64171779204934c17

**Review by ioapiset — APPROVED** [review]
LGTM.

**Pull request merged**